src/mui/detail/RecordForm.js

```diff
@@ -55,25 +55,32 @@ export const RecordForm = ({ children, handleSubmit, record, resource, basePath
         <div style={{ padding: '0 1em 1em 1em' }}>
             {React.Children.map(children, input => (
                 <div key={input.props.source} style={input.props.style}>
-                    { input.props.includesLabel ?
-                        <Field
-                            {...input.props}
-                            name={input.props.source}
-                            component={input.type}
-                            resource={resource}
-                            record={record}
-                            basePath={basePath}
-                        />
-                        :
-                        <Field
-                            {...input.props}
-                            name={input.props.source}
-                            component={Labeled}
-                            label={input.props.label}
-                            resource={resource}
-                            record={record}
-                            basePath={basePath}
-                        >{ input }</Field>
+                    { input.props.includesField === false ?
+                        (input.props.includesLabel ?
+                            <Field
```

Comment:
Can't we refactor all these props to ease readability, something like:

``` js
const fieldProps = {
    ...input.props,
    name: input.props.source,
    label: input.props.includesLabel ? input.props.label : undefined,
};

return <Field {...fieldProps} />;
```

Indeed, we change almost nothing and all these lines are quite intimidating. :)